### DISCUSSION TITLE
Buffering per service without Kafka

### DISCUSSION THREAD

**Discussion by brycefisher** [vectordotdev/vector#21664] [Q&A]
I'm interested in whether vector might be a good fit for my use case -- I have an existing solution, but it doesn't seem to solve a specific problem I have. I have a kubernetes cluster using the prometheus operator to scrape metrics from many different workloads. Unfortunately, if a single Deployment or other workload begins to create many pods rapidly, this can produce too many series for my metrics sink.

In my current solution, I don't have a good way to proactively limit the number of cumulative series from a single ServiceMonitor (all pods from a Deployment, for example). And no single pod produces _so many_ metrics that I can use prometheus `scrape_config.label_limits`.

If I used vector without kafka [as described in the docs](https://vector.dev/docs/setup/going-to-prod/arch/aggregator/#going-to-production), would it be possible to have vector drop metrics from a single ServiceMonitor based on the cumulative cardinality of that service monitor?

**Comment by jszwedko** [comment]
I think that might be possible using the [throttle transform](https://vector.dev/docs/reference/configuration/transforms/throttle/). That transform can be used to limit the number of events per time window for a given key.
